contracts/interchain-token-service/src/state.rs

```diff
@@ -107,6 +180,104 @@ pub fn load_all_its_contracts(
         .collect::<Result<HashMap<_, _>, _>>()?)
 }
 
+pub fn save_token_info(
+    storage: &mut dyn Storage,
+    directional_chain: DirectionalChain,
```

Comment:
For this function, I would just pass in both chains as `source` and `dest`, and create both of the necessary structs